tests/frameworks/function.py

```diff
@@ -0,0 +1,220 @@
+from collections import defaultdict, deque
+
+try:
+    from collections.abc import Iterator
+except ImportError:
+    from collections import Iterator  # type: ignore[attr-defined,no-redef]
+
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol  # type: ignore[misc]
+
+from os.path import abspath
+from types import FunctionType, ModuleType
+from typing import Any, Dict, List, Optional, Set, Tuple, Type, Union, cast
+
+from module import origin
+
+from bytecode import Bytecode
+
+
+def linenos(f):
+    # type: (FunctionType) -> Set[int]
+    """Get the line numbers of a function."""
+    try:
+        return {instr.lineno for instr in Bytecode.from_code(f.__code__) if hasattr(instr, "lineno")}
```

Comment:
To avoid issues early on in the collection it may be worth using dis here rather than bytecode.

Reply:
I think we can delete this altogether as we don't need line info here. I'll do some more clean-up.